### PR TITLE
[New 3D rendering mode] Anaglyph yellow/blue

### PR DESCRIPTION
**Pull request by Ferk**
Yellow/blue anaglyphs are one of the most recent iterations of the anaglyph systems for 3D vision. It's said to give the best results on proper light conditions and are already [supported by ffmpeg](https://github.com/FFmpeg/FFmpeg/blob/release/2.5/libavfilter/vf_stereo3d.c#L44).

These can be watched with amber-darkblue glasses and cover the only remaining combination of the 3 colors possible when used in anaglyphs based on complete color separation (cyan is blue+green, magenta is red+blue and yellow is red+green). It's natural to include it since we already support red-cyan and green-magenta.

There has been for a while a [feature request in the forums](http://forum.kodi.tv/showthread.php?tid=182597) asking for this.
## 

The pull request already adds the new entry in the list of 3D rendering methods in the GUI interface, however the menu entry doesn't have a translatable string id. This is my first involvement with this project and I'm not familiar with the way in which new translatable strings are added by id.

Please, could someone brief me on how to add a new string that can be translated by the po files?

EDIT: nevermind, I figured out where the id was coming from.

### CODE REVIEW

**Comment by FernetMenta** [comment]
check your commits. "merge branch" is not supposed to be included here

File: xbmc/cores/VideoRenderers/WinRenderer.cpp [hunk 1]
```diff
@@ -1042,7 +1042,8 @@ void CWinRenderer::RenderProcessor(DWORD flags)
   IDirect3DSurface9* target;
   if ( m_bUseHQScaler 
     || g_graphicsContext.GetStereoMode() == RENDER_STEREO_MODE_ANAGLYPH_RED_CYAN
-    || g_graphicsContext.GetStereoMode() == RENDER_STEREO_MODE_ANAGLYPH_GREEN_MAGENTA)
+    || g_graphicsContext.GetStereoMode() == RENDER_STEREO_MODE_ANAGLYPH_GREEN_MAGENTA
+	|| g_graphicsContext.GetStereoMode() == RENDER_STEREO_MODE_ANAGLYPH_YELLOW_BLUE)
```
Comment (MartijnKaijser) [inline xbmc/cores/VideoRenderers/WinRenderer.cpp]: Check indenting and few lines down as well

**Comment by Ferk** [comment]
Cleaned up.

Also the translatable string is now in place properly. I had to add a switch since the order of the list no longer matches the order in the translation strings.

Thank you very much for your guidance.

**Comment by fritsch** [comment]
@da-anda Perhaps we can "bikeshed" @Ferk to have a look at the switching problem?

@Ferk Thanks much. 

The rest looks "straight" forward. When I look through the special casing (not in your code) - it seems a little refactor could improve the overall situation.

File: xbmc/cores/VideoRenderers/RenderFlags.cpp [hunk 2]
```diff
@@ -101,6 +101,7 @@ namespace RenderManager {
       convert["anaglyph_cyan_red"]      = 0u;
       convert["right_left"]             = CONF_FLAGS_STEREO_MODE_SBS | CONF_FLAGS_STEREO_CADANCE_RIGHT_LEFT;
       convert["anaglyph_green_magenta"] = 0u;
+      convert["anaglyph_yellow_blue"] = 0u;
```
Comment (da-anda) [inline xbmc/cores/VideoRenderers/RenderFlags.cpp]: nitpick - could you bring the "=" sign in line with the others?

File: xbmc/guilib/StereoscopicsManager.cpp [hunk 3]
```diff
@@ -241,9 +243,28 @@ RENDER_STEREO_MODE CStereoscopicsManager::GetStereoModeOfPlayingVideo(void)
 
 const std::string &CStereoscopicsManager::GetLabelForStereoMode(const RENDER_STEREO_MODE &mode) const
 {
-  if (mode == RENDER_STEREO_MODE_AUTO)
-    return g_localizeStrings.Get(36532);
-  return g_localizeStrings.Get(36502 + mode);
+  int msgId;
+  switch(mode) {
```
Comment (da-anda) [inline xbmc/guilib/StereoscopicsManager.cpp]: I'm not sure I like this construct here. Having only one special case was fine, but having half the labels hardcoded while the rest is "dynamic" seems a bit confusing. Maybe we should have a label mapping struct for all modes to streamline this a bit? @fritsch, opinions?

**Comment by da-anda** [comment]
@Ferk - do you know if there are movies out there that are encoded using this format so that we also might need meta-data detection for it? Having it only as output mode is fine though - was just wondering. So besides of the comment on the GetLabelForStereoMode this PR looks fine form my side (assume it's working correctly on DirectX and OpenGL).

@popcornmix mind having a look on the changes to the mmal renderer?

**Comment by popcornmix** [comment]
Anaglyph is currently unsupported on Pi, so no change required here.
There was a test build with anaglyph support but it was implemented as a post-processing stage of decoder (like deinterlace), rather than a render stage.

**Comment by da-anda** [comment]
@Ferk - mind rebasing it on master?

**Comment by Ferk** [comment]
@da-anda rebased. 
Also I don't know about metadata detection for this type of anaglyph. Sorry for the late answer, I was having some personal stuff to deal with.

**Comment by da-anda** [comment]
thanks @Ferk - just tested on Windows (testbuild available on mirrors) and it seems to work correctly (don't have yellow/blue glasses - but the effect looked correct).

jenkins build this please

**Pull request merged**